XNor/functor

```diff
@@ -1,18 +1,24 @@
-  λ(a : Type)
-→ (   { map =
-            λ(b : Type)
-          → λ(c : Type)
-          → λ(f : b → c)
-          → λ(fa : ./type  a b)
-          →     let X = constructors (./type  a b)
-            
-            in  merge
-                { Both    =
-                      λ(xn : { head : a, tail : b })
-                    → X.Both { head = xn.head, tail = f xn.tail }
-                , Neither = X.Neither
-                }
-                fa
-      }
-    : ../functor  (./type  a)
-  )
+    let XNor = ./type
+
+in  let Functor = ../functor
+
+in    λ(a : Type)
+    → (   { map =
+                λ(b : Type)
+              → λ(c : Type)
+              → λ(f : b → c)
+              → λ(fa : XNor a b)
+              →     let X = constructors (XNor a c)
+
+                in  let handlers =
```

Comment:
Can you inline these handlers again? This indirection doesn’t actually clarify anything, IMO.